com.st.STM32CubeMX.metainfo.xml

```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+  <id>com.st.STM32CubeMX</id>
+  
+  <name>STM32CubeMX</name>
+  <summary>A graphical tool for configuring STM32 microcontrollers and microprocessors</summary>
+
+  <developer id="com.st">
+    <name>ST Microelectronics</name>
+  </developer>
+  
+  <metadata_license>MIT</metadata_license>
+  <project_license>LicenseRef-proprietary=https://www.st.com/resource/en/license_agreement/dm00218346.pdf</project_license>
```

Comment:
is this redistributable? If not you have to use extra-data. https://docs.flatpak.org/en/latest/module-sources.html#extra-data

Reply:
the license is also in /app/stm32cubemx/help/software_license_agreement.pdf

Reply:
does it say anywhere in the license that redistribution is allowed?

Reply:
> The redistribution,reproduction and use in source and binary forms of the software package or any part thereof, with or without modification, are permitted provided that the following conditions are met ...

Reply:
and these conditions are ?

Reply:
https://www.st.com/resource/en/license_agreement/dm00218346.pdf:
> Under STMicroelectronics’ intellectual property rights and subject to applicable licensing terms for any third-party software incorporated in this software package and applicable Open Source Terms (as defined here below), the redistribution, reproduction and use in source and binary forms of the software package or any part thereof, with or without modification, are permitted provided that the following conditions are met:
> 1. Redistribution of source code (modified or not) must retain any copyright notice, this list of conditions and the following disclaimer.
> 2. Redistributions in binary form, except as embedded into microcontroller or microprocessor device manufactured by or for STMicroelectronics or a software update for such device, must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
> 3. Neither the name of STMicroelectronics nor the names of other contributors to this software package may be used to endorse or promote products derived from this software package or part thereof without specific written permission.
> 4. This software package or any part thereof, including modifications and/or derivative works of this software package, must be used and execute solely and exclusively on or in combination with a microcontroller or a microprocessor devices manufactured by or for STMicroelectronics.
> 5. No use, reproduction or redistribution of this software package partially or totally may be done in any manner that would subject this software package to any Open Source Terms (as defined below).
> 6. Some portion of the software package may contain software subject to Open Source Terms (as defined below) applicable for each such portion (“Open Source Software”), as further specified in the software package. Such Open Source Software is supplied under the applicable Open Source Terms and is not subject to the terms and conditions of license hereunder. “Open Source Terms” shall mean any open source license which requires as part of distribution of software that the source code of such software is distributed therewith or otherwise made available, or open source license that substantially complies with the Open Source definition specified at www.opensource.org and any other comparable open source license such as for example GNU General Public License (GPL), Eclipse Public License (EPL), Apache Software License, BSD license and MIT license.
> 7. This software package may also include third party software as expressly specified in the software package subject to specific license terms from such third parties. Such third party software is supplied under such specific license terms and is not subject to the terms and conditions of license hereunder. By installing copying, downloading, accessing or otherwise using this software package, the recipient agrees to be bound by such license terms with regard to such third party software.
> 8. STMicroelectronics has no obligation to provide any maintenance, support or updates for the software package.
> 9. The software package is and will remain the exclusive property of STMicroelectronics and its licensors. The recipient will not take any action that jeopardizes STMicroelectronics and its licensors' proprietary rights or acquire any rights in the
> software package, except the limited rights specified hereunder.
> 10. The recipient shall comply with all applicable laws and regulations affecting the use of the software package or any part thereof including any applicable export control law or regulation.
> 11. Redistribution and use of this software package partially or any part thereof other than as permitted under this license is void and will automatically terminate your rights under this license

As far as I understand, these conditions allow for its redistribution.